torch/csrc/jit/python/python_arg_flatten.cpp

```diff
@@ -141,6 +166,21 @@ py::object unflatten_rec(
     return py::reinterpret_borrow<py::object>(THPUtils_packString(str));
   } else if (type == D::NoneType) {
     return py::reinterpret_borrow<py::object>(py::none());
+  } else if (type == D::Long) { // unwrap integers as variables for tracer
+    if (var_it == var_it_end)
+      throw std::runtime_error("Not enough Variables given to unflatten");
+    auto var = *var_it++;
+    return py::reinterpret_borrow<py::object>(THPVariable_Wrap(var));
+  } else if (type == D::Double) { // unwrap floats as variables for tracer
```

Comment:
Can we merge `Double` and `Bool` case?
Why for long, it is `reinterpret_borrow`, but for Bool, and Double, it is `reinterpret_steal`?

Reply:
Thanks for mentioning. I think borrow is safer as is does not change the ownership. I'll leave all to borrow.

Reply:
Can we do `type == D:Double || type ==  D::Bool` here?

Reply:
We can. Depending on how we want to handle these types.
Currently we're unwrapping all scalars + bool as a variable so that they're not traced as a constant but remain as a variable model input. With this is mind, we can handle double/long/bool similarly.

Reply:
it's also the same code for the last else branch. Maybe they can be merged with proper comments